src/main/java/de/dhbw/mh/slang/codegen/JavaBytecodeGenerator.java

```diff
@@ -59,12 +59,25 @@ public String visitPost( AstUnaryOperation node, String base ){
 
 	@Override
 	public String visitPost( AstBinaryOperation node, String lhs, String rhs ){
+		String prefix= "";
+		switch (node.getDatatype()) {
+			case BOOL: prefix = "i";break;
+			case I8: prefix = "i";break;
+			case I16: prefix = "i";break;
+			case I32: prefix = "i";break;
+			case I64: prefix = "l";break;
+			case F32: prefix = "f";break;
+			case F64: prefix = "d";break;		
+			default:
+				prefix = "i";
+				break;
+		}
 		switch( node.OPERATOR ){
 			case ADD:{
 				throw new RuntimeException( "not yet implemented" );
 			}
 			case SUBTRACT:{
-				throw new RuntimeException( "not yet implemented" );
+				return String.format("%s%s" + prefix + "sub%n", lhs, rhs);
```

Comment:
In my opinion, you should not mix concatenation and string formatting. This increases the complexity without any necessity.